### PR TITLE
102 anonymous compound access

### LINKED ISSUE #102: Anonymous Compound Access
**As a** anonymous user
**I want** to be able to read all contents of the compound api
**So that** I don't have to login to access the data

In general, GET requests in the Compound API should not require authentication

**Acceptance Criteria**
When I can perform GET on all the URL resources below including the paginated lists, the detail resources obtained by appending the ID to the URL, and any filtering/search functions.
http://api.chemreg.epa.gov/
http://api.chemreg.epa.gov/defined-compounds/
http://api.chemreg.epa.gov/ill-defined-compounds/
http://api.chemreg.epa.gov/query-structure-types/

### PR DESCRIPTION
**Pull request by AScaduto**
closes #102 and #117 

https://www.django-rest-framework.org/api-guide/permissions/#djangomodelpermissionsoranonreadonly

http://127.0.0.1:8000/#section/Authentication/anonAuth

### CODE REVIEW

File: chemreg/auth/tests/test_authentication.py [hunk 1]
```diff
@@ -40,3 +40,11 @@ def test_settings(settings):
         "rest_framework.authentication.BasicAuthentication"
         in settings.REST_FRAMEWORK["DEFAULT_AUTHENTICATION_CLASSES"]
     )
+
+    # Authenticated users can perform any request.
+    # Unauthorised users will only be permitted if the request
+    # method is one of the "safe" methods; GET, HEAD or OPTIONS.
+    assert (
+        "rest_framework.permissions.DjangoModelPermissionsOrAnonReadOnlyDjangoModelPermissionsOrAnonReadOnly"
```
Comment (debboutr) [inline chemreg/auth/tests/test_authentication.py:48]:
```suggestion
        "rest_framework.permissions.DjangoModelPermissionsOrAnonReadOnly"
```
hint: doubling the classname does NOT increase security

**Comment by debboutr** [comment]
is anyone else getting test failures here?
```
- chemreg/compound/tests/test_views.py:145 test_definedcompound_admin_user
- chemreg/compound/tests/test_views.py:199 test_query_structure_type_delete
- chemreg/compound/tests/test_views.py:218 test_deprecated_qst_in_illdefined
```
AFAIK, it's coming from the `has_permission` method of the `DjangoModelPermissions` class, where the User that we are getting from UserFactory doesn't have any permissions applied. Here is a sloppy way just to get the `test_definedcompound_admin_user` test to pass, I'm not sure of the best way to get the user permissions to be applied to the "faked" User in our fixtures or otherwise.
```
    from django.contrib.auth.models import Permission

    for p in Permission.objects.all():
        if p.name == "Can add defined compound":
            admin_user.user_permissions.add(p.pk)
```

more work may need to be done along these lines if someone else could confirm for me that their tests aren't passing here either. For instance, we can start using the `user` and `admin_user` from fixtures instead of using the `user_factory` and then building in the test
`def test_definedcompound_admin_user(admin_user, user, defined_compound_factory, client):`

File: chemreg/auth/tests/test_authentication.py (hunk 1 above)
Comment (AScaduto) [reply chemreg/auth/tests/test_authentication.py]: What about tripling it? Thanks for catching that lol

**Comment by AScaduto** [comment]
@debboutr with the revised permission "DjangoModelPermissionsOrAnonReadOnly" I do get those test failures. Initially I went with the permission "IsAuthenticatedOrReadOnly" which did not fail those tests but after chatting with @mcovalt, after #117 was brought up, I revised it to "DjangoModelPermissionsOrAnonReadOnly" so that this change would also close out #117 as well. 

I don't mind going back and revising those tests as a part of this issue or a new one if we can settle on a way to handle the test user permissions.

**Comment by AScaduto** [comment]
@debboutr All tests are now passing in my environment after commit 0578892; Looks great from my end

**Pull request merged**